### PR TITLE
Updates React to 0.14.0

### PR DESCRIPTION
**Pull request by nathanielparke**
This PR updates our react dependency to 0.14.0. It adds a few react dependencies that are necessary due to the composition of the react project. See https://facebook.github.io/react/blog/2015/10/07/react-v0.14.html

### CODE REVIEW

File: huxley/www/static/js/huxley/components/AdvisorAssignmentsView.js [hunk 1]
```diff
@@ -80,16 +80,18 @@ var AdvisorAssignmentsView = React.createClass({
...
-                <th>Committee</th>
-                <th>Country</th>
-                <th>Delegation Size</th>
-                <th>{finalized ?
-                  "" :
-                  "Delete Assignments"}
-                </th>
-              </tr>
-              {this.renderAssignmentRows()}
+              <tbody>
```
Comment (tdowds) [inline huxley/www/static/js/huxley/components/AdvisorAssignmentsView.js]: If this is wrapping around headers shouldn't we be using `<thead>`?

File: huxley/www/static/js/huxley/components/AdvisorProfileView.js [hunk 2]
```diff
@@ -61,169 +62,171 @@ var AdvisorProfileView = React.createClass({
...
-                  {'$' + school.fees_paid.toFixed(2)}
-                </td>
-              </tr>
-              <tr>
-                <td className="fieldLabel">Balance</td>
-                <td className="field">
-                  {'$' + (school.fees_owed - school.fees_paid).toFixed(2)}
-                </td>
-              </tr>
+              <tbody>
```
Comment (tdowds) [inline huxley/www/static/js/huxley/components/AdvisorProfileView.js]: Same as thing, should we be using `<thead>` here and then `<tbody>` for everything else? Or does that affect the appearance too adversely?

File: huxley/www/static/js/huxley/components/LoginView.js [hunk 3]
```diff
@@ -8,9 +8,12 @@
 var console = require('console');
 
 var $ = require('jquery');
-var React = require('react/addons');
+var classNames = require('classnames');
+var LinkedStateMixin = require('react-addons-linked-state-mixin');
+var React = require('react');
 var Router = require('react-router');
 
+
```
Comment (tdowds) [inline huxley/www/static/js/huxley/components/LoginView.js]: I don't think there should be an extra space here, should there?

File: package.json [hunk 4]
```diff
@@ -6,13 +6,17 @@
...
     "es6-promise": "2.0.0",
     "flux": "^2.1.1",
+    "fbjs": "0.8.3",
     "jquery": "2.1.1",
     "jquery-ui": "1.10.5",
     "js-cookie": "~2.0.3",
-    "react": "~0.13.3",
-    "react-router": "~0.13.0"
+    "react": "0.14.0",
+    "react-dom": "0.14.0",
```
Comment (tdowds) [inline package.json]: All of these dependencies are no longer part of the react package, you have to install them separately?

**Comment by tdowds** [comment]
Thanks for doing this Nate, I could see this being really annoying to do

File: package.json (hunk 4 above)
Comment (nathanielparke) [inline package.json]: yeah, React had a major restructuring. Read about it here: https://facebook.github.io/react/blog/2015/10/07/react-v0.14.html

**Comment by kmeht** [comment]
LGTM. If I had one nit, (and this is a weird one), it would be to keep the `classnames` require as cx(). It's a way shorter name for a commonly used function. e.g.
```js
var cx = require('classnames');
```

**Pull request merged**